shell/platform/android/io/flutter/embedding/engine/FlutterEnginePluginRegistry.java

```diff
@@ -259,16 +260,32 @@ private boolean isAttachedToActivity() {
   }
 
   @Override
-  public void attachToActivity(@NonNull Activity activity, @NonNull Lifecycle lifecycle) {
+  public void attachToActivity(
+      @NonNull Activity activity,
+      @NonNull Lifecycle lifecycle,
+      @NonNull PlatformViewsController platformViewsController
+  ) {
     Log.v(TAG, "Attaching to an Activity."
         + (isWaitingForActivityReattachment ? " This is after a config change." : ""));
     // If we were already attached to an Android component, detach from it.
     detachFromAndroidComponent();
 
     this.activity = activity;
-    this.activityPluginBinding = new FlutterEngineActivityPluginBinding(activity);
+    this.activityPluginBinding = new FlutterEngineActivityPluginBinding(
+        activity,
+        platformViewsController
+    );
     this.flutterEngineAndroidLifecycle.setBackingLifecycle(lifecycle);
 
+    // Activate the PlatformViewsController. This must happen before any plugins attempt
+    // to use it, otherwise an error strack trace will appear that says there is no
+    // flutter/platform_views channel.
+    platformViewsController.attach(
+        activity,
```

Comment:
So this part I understand. I'm not sure why anyone besides the fragment is involved in FlutterEngineActivityPluginRegistry.attachToActivity.

Reply:
Can you specify exactly what you mean? The reference to "anyone besides the fragment" leaves this ambiguous as to what you're getting at.

Reply:
Talked offline, resolving old comments.